spec/services/hyrax/works_count_service_spec.rb

```diff
@@ -28,28 +30,47 @@
 
     let(:access) { :edit }
     let(:documents) { [doc1, doc2] }
+    let(:connection) { instance_double(RSolr::Client) }
     let(:doc1) do
       SolrDocument.new(id: 'xyz123',
+                       title_tesim: ['test1'],
                        human_readable_type_tesim: ['Work type'],
                        visibility_ssi: 'restricted',
                        system_create_dtsi: '2015-01-01T20:50:35Z')
     end
     let(:doc2) do
       SolrDocument.new(id: 'yyx123',
                        human_readable_type_tesim: ['Work type'],
+                       title_tesim: ['test2'],
                        visibility_ssi: 'institution',
                        system_create_dtsi: '2015-02-01T20:50:35Z')
     end
 
-    let(:struct) { described_class::SearchResultForWorkCount }
+    let(:results) do
+      {
+        'response' =>
+            {
+              'docs' => documents,
+              'numFound' => 2
+            }
+      }
+    end
 
     before do
       allow(service).to receive(:search_results).and_return(documents)
+      allow(ActiveFedora::SolrService.instance).to receive(:conn).and_return(connection)
+      allow(connection).to receive(:get).with('select', params: { fq: '{!terms f=has_model_ssim}GenericWork',
+                                                                  rows: 10 }).and_return(results)
     end
 
-    context "when there are works in the collection" do
-      it "returns rows with document in the first column, last modified in the second column and integer count values in the third and fourth column" do
-        expect(subject).to eq [struct.new(doc1, '2015-01-01', 0, 'Work type', 'restricted'), struct.new(doc2, '2015-02-01', 0, 'Work type', 'institution')]
+    context 'when there are works in the collection' do
```

Comment:
What happens if there is an empty collection?

Reply:
I can add a test for that